MatrixSDK/Data/Store/MXFileStore/MXFileStore.m

```diff
@@ -823,9 +827,42 @@ - (MXMemoryRoomStore*)getOrCreateRoomStore:(NSString*)roomId
     MXFileRoomStore *roomStore = roomStores[roomId];
     if (nil == roomStore)
     {
-        // MXFileStore requires MXFileRoomStore objets
-        roomStore = [[MXFileRoomStore alloc] init];
-        roomStores[roomId] = roomStore;
+        NSString *roomFile = [self messagesFileForRoom:roomId forBackup:NO];
+        
+        if ([[NSFileManager defaultManager] fileExistsAtPath:roomFile])
+        {
+            MXFileRoomStore *roomStore;
+            @try
+            {
+                NSDate *startDate = [NSDate date];
+                roomStore = [NSKeyedUnarchiver unarchiveObjectWithFile:roomFile];
```

Comment:
It should be processed on a separated thread.

Reply:
It is now.